app/services/purl_and_stacks_service.rb

```diff
@@ -56,6 +56,18 @@ def delete(version:)
     UpdatePurlMetadataService.new(purl).delete! if legacy_purl_enabled?
   end
 
+  # Delete the PURL and Stacks files.
+  # @param version [String] the version number
+  def withdraw(version:)
+    return unless versioned_files_enabled? && already_versioned_layout?
```

Comment:
I think this should raise. Trying to withdraw an unversioned object indicates that something is wrong.